src/PrestaShopBundle/Form/Admin/Sell/Product/EventListener/ProductTypeListener.php

```diff
@@ -83,12 +83,28 @@ public function adaptProductForm(FormEvent $event): void
         }
 
         if (ProductType::TYPE_COMBINATIONS === $productType) {
-            $this->removeSuppliers($form);
+            $this->removeProductSuppliers($form);
             $this->removeStock($form);
         } elseif ($initialProductType !== ProductType::TYPE_COMBINATIONS) {
             $this->removeCombinations($form);
         }
 
+        $optionsField = $form->get('options');
+
+        // As this is also executed on pre-submit, we need to verify that suppliers exist before removing it
+        if ($optionsField->has('suppliers')) {
+            $suppliers = $optionsField->get('suppliers')->get('supplier_ids')->getConfig()->getOptions()['choices'];
+
+            // Don't display the suppliers part if there are no suppliers
+            if (count($suppliers) <= 0) {
+                $this->removeSuppliers($form);
+
+                if ($optionsField->has('product_suppliers')) {
```

Comment:
Maybe this should be placed in the `removeProductSuppliers` instead, I don't know if calling remove on a non-existent field triggers an error